### ISSUE TITLE
[Feature request] Crawl delay based on mimetype

### ISSUE THREAD

**Issue by OkkeKlein** [Norconex/collector-http#246]
For my use case I could use a setting in GenericDelayResolver to crawl certain contenttype or urls with different delays. So more like GenericRecrawlableResolver.

So the dynamic XML feeds I crawl have more delay to keep the server from crashing and the PDF's it links to have less delay as they are static.

**Comment by essiembre** [comment]
I do not see how we can implement this given the mime type is not known at that stage.  The delay is the time to wait between HTTP requests, but the mime-type can only be obtained AFTER the HTTP request was made.   

The recrawlable resolver works with mime types because we know it from a previous crawl (it is stored in the crawl store).   With the delay resolver, it would only work for re-crawls, but it is meant to work all the time, not just for recrawls.

It could be done to work on URL patterns though, since that we obviously have.  Could that work for you?

**Comment by OkkeKlein** [comment]
Makes sense. Yeah, URL regex pattern would work just fine.

**Comment by essiembre** [comment]
A new [ReferenceDelayResolver](http://www.norconex.com/collectors/collector-http/latest/apidocs/com/norconex/collector/http/delay/impl/ReferenceDelayResolver.html) was created to support delays based on document reference patterns.

Here is a usage sample (refer to the above javadoc link for all configuration options): 

```xml
      <delay class="com.norconex.collector.http.delay.impl.ReferenceDelayResolver"
              default="2000" 
              ignoreRobotsCrawlDelay="true"
              scope="crawler" >
          <pattern delay="5000">.*doc$</pattern>
          <pattern delay="10000">.*pdf$</pattern>
      </delay>
```
It is part of the [latest snapshot](http://www.norconex.com/collectors/collector-http/download).

Please let me know if that works for you.

**Comment by essiembre** [comment]
This fix was released in 2.5.0.
